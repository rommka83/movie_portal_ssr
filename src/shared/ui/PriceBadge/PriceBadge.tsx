import React, { HTMLAttributes } from 'react';
import styles from './priceBadge.module.css';
import cn from 'classnames';

interface IProps {
  price: boolean;
}

type props = HTMLAttributes<HTMLSpanElement> & IProps;

export const PriceBadge = ({ price, className }: props) => {
  return (
    <span className={cn(styles.priceBadge, price ? styles.pink : styles.gray, className)}>
      {price ? 'Подписка' : 'Бесплатно'}
    </span>
  );
};
