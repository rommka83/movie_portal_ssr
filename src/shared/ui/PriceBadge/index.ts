export { PriceBadge as default } from "./PriceBadge";
