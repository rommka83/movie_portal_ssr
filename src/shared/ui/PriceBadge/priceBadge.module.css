.priceBadge {
  font-size: 13px;
  line-height: 16px;
  font-weight: 400;
}

.pink {
  color: var(--pink);
}

.gray {
  color: var(--gray-light);
}
