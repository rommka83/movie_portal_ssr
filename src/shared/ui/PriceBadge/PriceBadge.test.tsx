import renderer from 'react-test-renderer';
import { PriceBadge } from './PriceBadge';

describe('проверка PriceBadge', () => {
  test('бесплатно', () => {
    const component = renderer.create(<PriceBadge price={false} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('подписка', () => {
    const component = renderer.create(<PriceBadge price />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
