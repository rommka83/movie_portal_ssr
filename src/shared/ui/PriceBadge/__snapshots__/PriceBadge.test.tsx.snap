// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка PriceBadge бесплатно 1`] = `
<span
  className="priceBadge gray"
>
  Бесплатно
</span>
`;

exports[`проверка PriceBadge подписка 1`] = `
<span
  className="priceBadge pink"
>
  Подписка
</span>
`;
